aten/src/ATen/native/NaiveConvolutionTranspose2d.cpp

```diff
@@ -291,98 +290,97 @@ void slow_conv_transpose2d_out_cpu_template(
+
+          // column-major matrices
+          cpublas::gemm(
+              TransposeType::NoTranspose,
+              TransposeType::Transpose,
+              m,
               n,
-              weight_.data_ptr<scalar_t>(),
+              k,
+              static_cast<scalar_t>(1),
+              input_n.data_ptr<scalar_t>(),
               m,
-              0,
-              columns.data_ptr<scalar_t>(),
-              n);
-
-          // Unpack columns back into input:
-          col2im<scalar_t>(
-              columns.data_ptr<scalar_t>(),
-              n_output_plane,
-              output_height,
-              output_width,
-              input_height,
-              input_width,
-              kernel_height,
-              kernel_width,
-              pad_height,
-              pad_width,
-              stride_height,
-              stride_width,
-              dilation_height,
-              dilation_width,
-              output_n.data_ptr<scalar_t>());
-
-          // Do Bias after:
-          // M,N,K are dims of matrix A and B
-          // (see http://docs.nvidia.com/cuda/cublas/#cublas-lt-t-gt-gemm)
-          int64_t m_ = n_output_plane;
-          int64_t n_ = output_height * output_width;
-          int64_t k_ = 1;
-
-          // Do GEMM (note: this is a bit confusing because gemm assumes
-          // column-major matrices)
-          if (bias.defined()) {
-            cpublas::gemm(
-                TransposeType::Transpose,
-                TransposeType::NoTranspose,
-                n_,
-                m_,
-                k_,
-                1,
-                ones.data_ptr<scalar_t>(),
-                k_,
-                bias_.data_ptr<scalar_t>(),
-                k_,
-                1,
-                output_n.data_ptr<scalar_t>(),
-                n_);
-          }
+              weight_.data_ptr<scalar_t>(),
+              n,
+              static_cast<scalar_t>(0),
+              columns_n.data_ptr<scalar_t>(),
+              m);
         }
 
-        // Resize output
-        if (is_batch) {
-          output.resize_({n_output_plane, output_height, output_width});
-          input_.resize_({n_input_plane, input_height, input_width});
-        }
-      });
+        // Unpack columns back into input:
+        col2im<scalar_t>(
+            columns_n.data_ptr<scalar_t>(),
+            n_output_plane,
+            output_height,
+            output_width,
+            input_height,
+            input_width,
+            kernel_height,
+            kernel_width,
+            pad_height,
+            pad_width,
+            stride_height,
+            stride_width,
+            dilation_height,
+            dilation_width,
+            output_n.data_ptr<scalar_t>(),
+            use_channels_last);
+      }
+    });
+  });
+
+  if (bias.defined()) {
+    output.add_(bias_.reshape({-1, 1, 1}));
+  }
+
+  // Resize output
+  if (is_batch) {
+    output.resize_({n_output_plane, output_height, output_width});
```

Comment:
It's not clear to me why you bothered optimizing the naive convolution transpose lol

Reply:
From the functional respective, we want to make sure a layer with Channels last support would have the same memory format propagation behavior `with` or `without` mkldnn. Just adding some fundamental optimization for the naive path, did not take too much time.